targets/TARGET_Cypress/TARGET_PSOC6/TARGET_CY8CPROTO_064_SB/device/COMPONENT_CM4/TOOLCHAIN_ARM/cyb06xx7_cm4_dual.sct

```diff
@@ -168,15 +172,26 @@ LR_IROM1 FLASH_START FLASH_SIZE
     }
 
     ; Application heap area (HEAP)
-    ARM_LIB_HEAP  +0 EMPTY RAM_START+RAM_SIZE-STACK_SIZE-ImageLimit(RW_IRAM1)
+    ARM_LIB_HEAP  +0 ALIGN 4 EMPTY RAM_START+RAM_SIZE-MBED_BOOT_STACK_SIZE-MBED_DATA_SHARED_SIZE-ImageLimit(RW_IRAM1)
     { 
     }
-    
+
     ; Stack region growing down
-    ARM_LIB_STACK RAM_START+RAM_SIZE EMPTY -STACK_SIZE 
+    ARM_LIB_STACK RAM_START+RAM_SIZE-MBED_DATA_SHARED_SIZE ALIGN 4 EMPTY -MBED_BOOT_STACK_SIZE 
     {
     }
 
+    ; region RAM overflowed with Stack area
+    ScatterAssert(ImageBase(ARM_LIB_STACK) + ImageLength(ARM_LIB_STACK) == RAM_START+RAM_SIZE-MBED_DATA_SHARED_SIZE)
```

Comment:
According to https://github.com/ARMmbed/mbed-os/pull/11998/commits/ac0c080cfbac6be983e2eac5ec281c83468d6c63#diff-23cb3f83d66b59da5638fbdc1aa09bddR180 `ImageBase(ARM_LIB_STACK)` should be equal to `RAM_START+RAM_SIZE-MBED_DATA_SHARED_SIZE`. Therefore this condition will never be met.

Reply:
This is the top of the stack or where the stack end. We are checking the stack boundary here.